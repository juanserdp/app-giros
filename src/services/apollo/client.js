import {ApolloClient, HttpLink, InMemoryCache} from '@apollo/client';

const client = new ApolloClient({
    link: new HttpLink({
        uri: 'http://localhost:4000/graphql'
    }),
    cache: new InMemoryCache(),
    onError: ({networkError, graphQLErrors}) => {
        console.log('graphQLError: ', graphQLErrors);
        console.log('networkError: ', networkError);
    }
});

export default client;