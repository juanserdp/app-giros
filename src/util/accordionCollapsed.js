export const accordionCollapsed = () => {
    const botones = document.getElementsByClassName("accordion-button collapsed");
    for (let boton of botones) {
        boton.style.backgroundColor = "#0d6efd";
        boton.style.borderTopLeftRadius = "0px";
        boton.style.borderTopRightRadius = "0px";
        boton.style.border = "0px";
        boton.style.height = "20px";
        boton.style.outlineStyle = "none";
        boton.style.color = "white";
        boton.style.fontFamily = "'Roboto Slab', serif";
        boton.style.fontWeight = "500";
        boton.style.fontSize = "1.5rem";
        boton.style.padding = "5px";
        boton.style.textAlign = "center"
    }
}