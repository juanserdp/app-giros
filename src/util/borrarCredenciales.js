export function borrarCredenciales(){
    localStorage.removeItem('numeroDocumento');
    localStorage.removeItem('clave');
    localStorage.removeItem('recordarCredenciales');
}