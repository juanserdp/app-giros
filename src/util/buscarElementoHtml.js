export function buscarTagHtml(html, tag) {
    let tagsHtml = [];
    function recolecarTagsHtml(html) {
        if (typeof html == "object") {
            if (html.type === tag) {
                tagsHtml.push(html);
            }
            if (html.children) {
                let children = html.children;
                if (children.length == 0) {
                    return;
                }
                else if (children.length == 1) {
                    html = children[0];
                    return recolecarTagsHtml(html);
                }
                else if (children.length > 1) {
                    children.map((nodo) => {
                        html = nodo;
                        return recolecarTagsHtml(html);
                    })
                }
            }
        }
        else console.log("No ha proporcionado una estructura html valida")
    }
    recolecarTagsHtml(html);
    return tagsHtml;
}