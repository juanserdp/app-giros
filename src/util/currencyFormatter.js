export const currencyFormatter = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    maximumFractionDigits: 0,
    minimumFractionDigits: 0
});

export const currencyFormatterWithDecimals = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD'
});