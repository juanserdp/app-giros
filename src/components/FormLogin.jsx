import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Container from 'react-bootstrap/Container';
import swal from 'sweetalert';
import { guardarCredenciales } from "../util/guardarCredenciales";
import { useState } from 'react';
import { borrarCredenciales } from '../util/borrarCredenciales';
import { useNavigate } from 'react-router-dom';
import { useMutation } from '@apollo/client';
import { LOGIN } from '../services/apollo/gql/login';

export function FormLogin() {
    const navigate = useNavigate();
    const estadoInicial = {
        numeroDocumento: localStorage.getItem("numeroDocumento") || "",
        clave: localStorage.getItem("clave") || ""
    }
    const [formLogin, setFormLogin] = useState(estadoInicial);
    const [recordarCredenciales, setRecordarCredenciales] = useState(localStorage.getItem("recordarCredenciales") || false);

    const [login] = useMutation(LOGIN, {
        variables: formLogin,
        onCompleted: ({ login }) => {
            if (login.token) {
                localStorage.setItem("jwt", login.token);
                navigate('/inicio');
            }
            else if (login.error) {
                console.error("Error!", login.error, "from FormLogin.js");
                swal("Error al iniciar sesión", login.error, "error");
            }
        },
        onError: ({ networkError }) => {
            if (networkError) {
                console.error(`[Network error]: ${networkError}`, " from FormLogin.js");
                if (networkError.result.errors[0].message)
                    swal("Error!", networkError.result.errors[0].message, "error");
            }
        }
    });
    const handlerSubmit = () => {
        if (recordarCredenciales) guardarCredenciales(formLogin.numeroDocumento, formLogin.clave);
        else borrarCredenciales();
        if (formLogin.numeroDocumento !== "" && formLogin.clave !== "") login();
        else swal("Error", "¡Todos los campos son obligatorios!", "error");
    }
    return (
        <Container fluid="md">
            <Form style={{
                width: "100%",
                height: "300px",
                maxWidth: "330px",
                padding: "15px",
                margin: "auto",
                marginTop: "calc(50vh - 150px)"
            }}>
                <Form.Group className="mb-3" controlId="formBasicEmail">
                    <Form.Label>Numero de Identificacion</Form.Label>
                    <Form.Control onChange={event => setFormLogin({ ...formLogin, numeroDocumento: event.target.value })
                    } value={formLogin.numeroDocumento} type="text" placeholder="Ingrese el numero" />
                </Form.Group>

                <Form.Group className="mb-3" controlId="formBasicPassword">
                    <Form.Label>Contraseña</Form.Label>
                    <Form.Control onChange={event => setFormLogin({ ...formLogin, clave: event.target.value })
                    } value={formLogin.clave} type="password" placeholder="Ingrese la contraseña" />
                </Form.Group>

                <Form.Group className="mb-3" controlId="formBasicCheckbox">
                    <Form.Check onChange={() => setRecordarCredenciales(!recordarCredenciales)
                    } defaultChecked={recordarCredenciales} type="checkbox" label="Recordarme" />
                </Form.Group>

                <Button onClick={(evento)=>handlerSubmit(evento)} variant="primary" >
                    Ingresar
                </Button>
            </Form>
        </Container>
    );
}


