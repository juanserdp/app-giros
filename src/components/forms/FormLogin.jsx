import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import Container from 'react-bootstrap/Container';
import swal from 'sweetalert';
import { guardarCredenciales } from "../../util/guardarCredenciales";
import { useState } from 'react';
import { borrarCredenciales } from '../../util/borrarCredenciales';
import { useNavigate } from 'react-router-dom';
import { useMutation } from '@apollo/client';
import { LOGIN } from '../../services/apollo/gql/login';
import { Spinner } from 'react-bootstrap';

export function FormLogin() {
    // INSTANCIAS DE CLASE
    const navigate = useNavigate();

    // CONSTANTES
    const estadoInicial = {
        numeroDocumento: localStorage.getItem("numeroDocumento") || "",
        clave: localStorage.getItem("clave") || ""
    };

    // ESTADOS
    const [formLogin, setFormLogin] = useState(estadoInicial);
    const [recordarCredenciales, setRecordarCredenciales] = useState(localStorage.getItem("recordarCredenciales") || false);

    const [login, { loading }] = useMutation(LOGIN, {
        variables: formLogin,
        onCompleted: ({ login }) => {
            if (login.token) {
                localStorage.setItem("jwt", login.token);
                navigate('/inicio');
            }
            else if (login.error) {
                console.error("Error!", login.error, "from FormLogin.js");
                swal("Error al iniciar sesión", login.error, "error");
            }
        },
        onError: ({ networkError }) => {
            if (networkError) {
                console.error(`[Network error]: ${networkError}`, " from FormLogin.js");
                if (networkError.result.errors[0].message)
                    swal("Error!", networkError.result.errors[0].message, "error");
            }
        }
    });
    const handlerSubmit = () => {
        if (recordarCredenciales) guardarCredenciales(formLogin.numeroDocumento, formLogin.clave);
        else borrarCredenciales();
        if (formLogin.numeroDocumento !== "" && formLogin.clave !== "") login();
        else swal("Error", "¡Todos los campos son obligatorios!", "error");
    }
    return (
        <Container fluid="md">

            <Form style={{
                width: "100%",
                height: "650px",
                maxWidth: "330px",
                padding: "15px",
                margin: "auto",
                marginTop: "calc(50vh - 325px)"
            }}>
                <img
                    src='https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQy5C__8LdlPzxa7WMOTs54k7jwNlRyKS4xKzIAN0IL2skfCe_yeOQqag5wSoYkwGYqoqo&usqp=CAU'
                    style={{
                        width: "100%",
                        height: "300px"
                    }}
                ></img>
                <Form.Group className="mb-3" controlId="formBasicEmail">
                    <Form.Label>Numero de Identificacion</Form.Label>
                    <Form.Control onChange={event => setFormLogin({ ...formLogin, numeroDocumento: event.target.value })
                    } value={formLogin.numeroDocumento} type="text" placeholder="Ingrese el numero" />
                </Form.Group>

                <Form.Group className="mb-3" controlId="formBasicPassword">
                    <Form.Label>Contraseña</Form.Label>
                    <Form.Control onChange={event => setFormLogin({ ...formLogin, clave: event.target.value })
                    } value={formLogin.clave} type="password" placeholder="Ingrese la contraseña" />
                </Form.Group>

                <Form.Group className="mb-3" controlId="formBasicCheckbox">
                    <Form.Check onChange={() => setRecordarCredenciales(!recordarCredenciales)
                    } defaultChecked={recordarCredenciales} type="checkbox" label="Recordarme" />
                </Form.Group>
                {/* {(loading) ? (
                    <Button variant="primary" disabled>

                    </Button>
                ) : (
                    <Button onClick={(evento) => handlerSubmit(evento)} variant="primary" >
                        Ingresar
                    </Button>
                )} */}

                <Button disabled={loading ? true : false} onClick={(evento) => handlerSubmit(evento)} variant="primary"  >
                    {loading ? (<><Spinner
                        as="span"
                        animation="grow"
                        size="sm"
                        role="status"
                        aria-hidden="true"
                    /> &nbsp;
                        Cargando...</>) : "Ingresar"}
                </Button>

            </Form>
        </Container>
    );
}


