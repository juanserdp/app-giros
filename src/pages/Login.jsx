import { FormLogin } from "../components/forms/FormLogin";

export default function Login() {
    return <FormLogin />;
};