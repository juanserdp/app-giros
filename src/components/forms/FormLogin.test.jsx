import TestRenderer from 'react-test-renderer';
import { FormLogin } from './FormLogin';
import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import Login from '../../pages/Login';
import { ApolloClient, ApolloProvider, InMemoryCache, createHttpLink } from "@apollo/client";
import { buscarTagHtml } from '../../util/buscarElementoHtml';

const httpLink = createHttpLink({
    uri: 'http://192.168.0.24:4000/graphql'
});

const client = new ApolloClient({
    cache: new InMemoryCache(),
    link: httpLink,
});
describe("FormLogin", () => {
    it('Muestra un modal de error al pasarle una contraseña mala', () => {
        const component = TestRenderer.create(
            <ApolloProvider client={client}>
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={
                            <Login>
                                <FormLogin />
                            </Login>
                        } />
                    </Routes>
                </BrowserRouter>
            </ApolloProvider>
            ,
        );
        let tree = component.toJSON();
        // RECORDARME -> INGRESAR
        TestRenderer.act(() => {
            const inputs = buscarTagHtml(tree, "input");
            const checkbox = inputs.filter(i => i.props.type === "checkbox");
            console.log(checkbox)
            checkbox[0].props.defaultChecked = true;
        })
        TestRenderer.act(() => {
            const button = buscarTagHtml(tree, "button");
            button[0].props.onClick();
        });
        expect(tree).toMatchSnapshot();

        // NO RECORDARME -> INGRESAR
        TestRenderer.act(() => {
            const inputs = buscarTagHtml(tree, "input");
            const checkbox = inputs.filter(i => i.props.type === "checkbox");
            console.log(checkbox)
            checkbox[0].props.defaultChecked = false;
        })
        TestRenderer.act(() => {
            const button = buscarTagHtml(tree, "button");
            button[0].props.onClick();
        });
        expect(tree).toMatchSnapshot();
    });
    it('Muestra un modal de error al pasarle una contraseña mala', () => {
        const component = TestRenderer.create(
            <ApolloProvider client={client}>
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={
                            <Login>
                                <FormLogin />
                            </Login>
                        } />
                    </Routes>
                </BrowserRouter>
            </ApolloProvider>
            ,
        );
        let tree = component.toJSON();
        // NO RECORDARME -> INGRESAR
        TestRenderer.act(() => {
            const inputs = buscarTagHtml(tree, "input");
            const checkbox = inputs.filter(i => i.props.type === "checkbox");
            console.log(checkbox)
            checkbox[0].props.defaultChecked = false;
        })
        TestRenderer.act(() => {
            const button = buscarTagHtml(tree, "button");
            button[0].props.onClick();
        });
        expect(tree).toMatchSnapshot();
    });
})
