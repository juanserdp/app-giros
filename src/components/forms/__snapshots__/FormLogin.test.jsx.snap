// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin Muestra un modal de error al pasarle una contraseña mala 1`] = `
<div
  className="container-md"
>
  <form
    className=""
    style={
      Object {
        "height": "650px",
        "margin": "auto",
        "marginTop": "calc(50vh - 325px)",
        "maxWidth": "330px",
        "padding": "15px",
        "width": "100%",
      }
    }
  >
    <img
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQy5C__8LdlPzxa7WMOTs54k7jwNlRyKS4xKzIAN0IL2skfCe_yeOQqag5wSoYkwGYqoqo&usqp=CAU"
      style={
        Object {
          "height": "300px",
          "width": "100%",
        }
      }
    />
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Numero de Identificacion
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        onChange={[Function]}
        placeholder="Ingrese el numero"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Contraseña
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        onChange={[Function]}
        placeholder="Ingrese la contraseña"
        type="password"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <div
        className="form-check"
      >
        <input
          className="form-check-input"
          defaultChecked={true}
          disabled={false}
          id="formBasicCheckbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="formBasicCheckbox"
          title=""
        >
          Recordarme
        </label>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Ingresar
    </button>
  </form>
</div>
`;

exports[`FormLogin Muestra un modal de error al pasarle una contraseña mala 2`] = `
<div
  className="container-md"
>
  <form
    className=""
    style={
      Object {
        "height": "650px",
        "margin": "auto",
        "marginTop": "calc(50vh - 325px)",
        "maxWidth": "330px",
        "padding": "15px",
        "width": "100%",
      }
    }
  >
    <img
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQy5C__8LdlPzxa7WMOTs54k7jwNlRyKS4xKzIAN0IL2skfCe_yeOQqag5wSoYkwGYqoqo&usqp=CAU"
      style={
        Object {
          "height": "300px",
          "width": "100%",
        }
      }
    />
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Numero de Identificacion
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        onChange={[Function]}
        placeholder="Ingrese el numero"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Contraseña
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        onChange={[Function]}
        placeholder="Ingrese la contraseña"
        type="password"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <div
        className="form-check"
      >
        <input
          className="form-check-input"
          defaultChecked={false}
          disabled={false}
          id="formBasicCheckbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="formBasicCheckbox"
          title=""
        >
          Recordarme
        </label>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Ingresar
    </button>
  </form>
</div>
`;

exports[`FormLogin Muestra un modal de error al pasarle una contraseña mala 3`] = `
<div
  className="container-md"
>
  <form
    className=""
    style={
      Object {
        "height": "650px",
        "margin": "auto",
        "marginTop": "calc(50vh - 325px)",
        "maxWidth": "330px",
        "padding": "15px",
        "width": "100%",
      }
    }
  >
    <img
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQy5C__8LdlPzxa7WMOTs54k7jwNlRyKS4xKzIAN0IL2skfCe_yeOQqag5wSoYkwGYqoqo&usqp=CAU"
      style={
        Object {
          "height": "300px",
          "width": "100%",
        }
      }
    />
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicEmail"
      >
        Numero de Identificacion
      </label>
      <input
        className="form-control"
        id="formBasicEmail"
        onChange={[Function]}
        placeholder="Ingrese el numero"
        type="text"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <label
        className="form-label"
        htmlFor="formBasicPassword"
      >
        Contraseña
      </label>
      <input
        className="form-control"
        id="formBasicPassword"
        onChange={[Function]}
        placeholder="Ingrese la contraseña"
        type="password"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <div
        className="form-check"
      >
        <input
          className="form-check-input"
          defaultChecked={false}
          disabled={false}
          id="formBasicCheckbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="formBasicCheckbox"
          title=""
        >
          Recordarme
        </label>
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Ingresar
    </button>
  </form>
</div>
`;
